packages/block-editor/src/components/media-placeholder/index.native.js

```diff
@@ -51,40 +63,69 @@ function MediaPlaceholder( props ) {
 	const emptyStateContainerStyle = useStyle( styles.emptyStateContainer, styles.emptyStateContainerDark );
 	const emptyStateTitleStyle = useStyle( styles.emptyStateTitle, styles.emptyStateTitleDark );
 
+	const renderContent = () => {
+		if ( isAppender === undefined || ! isAppender ) {
+			return (
+				<>
+					<View style={ styles.modalIcon }>
+						{ icon }
+					</View>
+					<Text style={ emptyStateTitleStyle }>
+						{ placeholderTitle }
+					</Text>
+					<Text style={ styles.emptyStateDescription }>
+						{ instructions }
+					</Text>
+				</>
+			);
+		} else if ( isAppender && disableMediaButtons ) {
+			return (
+				<Dashicon
+					icon="plus-alt"
+					style={ styles.addBlockButton }
+					color={ styles.addBlockButton.color }
+					size={ styles.addBlockButton.size }
+				/>
+			);
+		}
+	};
+
+	if ( isAppender && ! disableMediaButtons ) {
+		return null;
```

Comment:
it looks like there's sth wrong here, it is returning null if disableMediaButtons is undefined or false.